modules/common/test/helpers/statefulset.go

```diff
@@ -82,6 +83,9 @@ func (tc *TestHelper) SimulateStatefulSetReplicaReadyWithPods(name types.Namespa
 		for i := range pod.Spec.Containers {
 			pod.Spec.Containers[i].VolumeMounts = []corev1.VolumeMount{}
 		}
+		for i := range pod.Spec.InitContainers {
+			pod.Spec.InitContainers[i].VolumeMounts = []corev1.VolumeMount{}
+		}
```

Comment:
could you please do this in SimulateDeploymentReadyWithPods as well? Then I can drop https://github.com/openstack-k8s-operators/lib-common/pull/302